xpriv/xpriv.go

```diff
@@ -0,0 +1,139 @@
+package xpriv
+
+import (
+	"fmt"
+
+	"github.com/libsv/go-bk/bip32"
+	"github.com/libsv/go-bk/bip39"
+	"github.com/libsv/go-bk/chaincfg"
+)
+
+// Keys is a struct containing the xpriv, xpub and mnemonic
+type Keys struct {
+	xpriv    string
+	xpub     PublicKey
+	mnemonic string
+}
+
+// PublicKey is a struct containing public key information
+type PublicKey struct {
+	Xpub string
+}
+
+// Key represents basic key methods
+type Key interface {
+	String() string
+	XPub() PubKey
+}
+
+// PubKey represents public key methods
+type PubKey interface {
+	String() string
+}
+
+// KeyWithMnemonic represents methods for generated keys
+type KeyWithMnemonic interface {
+	Key
+	Mnemonic() string
+}
+
+// XPub return hierarchical struct which contain xpub info
+func (k *Keys) XPub() PubKey {
+	return k.xpub
+}
+
+// String return hierarchical deterministic private key
+func (k *Keys) String() string {
+	return k.xpriv
+}
```

Comment:
I feel this is wrong for some reason 😄 The method that returns xPriv should be called XPriv(), no? :D String suggests you're producing a string from the entire struct

Reply:
You are right, I changed it into XPriv() method